WordPress/gutenberg
# Navigation: Use the ListView in the Navigation block inspector controls

## What?
This replaces the OffCanvasEditor with the ListView component in the inspector controls for the Navigation block

## Why?
We shouldn't have two components that do the same thing.

## How?
Implements the new props we have added.

Also adds a wrapping div to allow the list view to scroll horizontally.

## Testing Instructions
1. Open the Site Editor 
2. Select the navigation block
3. Open the inspector controls
4. Confirm that the list view for the navigation block works the same as before

### Testing Instructions for Keyboard


## Screenshots or screencast 
<img width="315" alt="Screenshot 2023-04-17 at 15 37 57" src="https://user-images.githubusercontent.com/275961/232521824-4e37bab5-d1bc-4772-b147-2e1ef6946c2a.png">

Horizontal scrollbar:
<img width="328" alt="Screenshot 2023-04-17 at 15 38 04" src="https://user-images.githubusercontent.com/275961/232521779-7107cdde-2f07-437c-bdc2-57ea80d3b6d8.png">

Closes https://github.com/WordPress/gutenberg/issues/49986
